applications/GeoMechanicsApplication/custom_processes/geo_extrapolate_integration_point_values_to_nodes_process.h

```diff
@@ -0,0 +1,131 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ \.
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//                   Jonathan Nuttall
+//                   Wijtze Pieter Kikstra
+//                   Richard Faasse
+
+#pragma once
+
+#include "geometries/geometry.h"
+#include "includes/element.h"
+#include "includes/kratos_parameters.h"
+#include "includes/model_part.h"
+#include "includes/node.h"
+#include "processes/process.h"
+
+namespace Kratos
+{
+
+class Model;
+class NodalExtrapolator;
+class ProcessInfo;
+
+/**
+ * @class GeoExtrapolateIntegrationPointValuesToNodesProcess
+ * @ingroup GeoMechanicsApplication
+ * @brief This process extrapolates vales from the integration points to the nodes
+ * @details This process solves local problems in order to extrapolate the values from the gauss point to the nodes. Uses inverse for same number of nodes and GP and generalized inverse for cases where the number of GP in higher than the number of nodes
+ * Using as main reference: https://www.colorado.edu/engineering/CAS/courses.d/IFEM.d/IFEM.Ch28.d/IFEM.Ch28.pdf (Felippa Stress Recovery course)
+ * @author Vicente Mataix Ferrandiz
+ */
+class KRATOS_API(GEO_MECHANICS_APPLICATION) GeoExtrapolateIntegrationPointValuesToNodesProcess : public Process
+{
+public:
+    using GeometryType = Geometry<Node>;
+    using SizeType     = std::size_t;
+    using IndexType    = std::size_t;
+
+    KRATOS_CLASS_POINTER_DEFINITION(GeoExtrapolateIntegrationPointValuesToNodesProcess);
+
+    explicit GeoExtrapolateIntegrationPointValuesToNodesProcess(Model& rModel,
+                                                                Parameters ThisParameters = Parameters(R"({})"));
+    explicit GeoExtrapolateIntegrationPointValuesToNodesProcess(ModelPart& rMainModelPart,
+                                                                Parameters rParameters = Parameters(R"({})"));
```

Comment:
To be consistent with the previous constructor:
```suggestion
                                                                Parameters ThisParameters = Parameters(R"({})"));
```

Reply:
Done